tests/integration/unsafe_snapshot_tests/bad_resource/resource.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright 2016-2022, Pulumi Corporation.  All rights reserved.
+
+import * as pulumi from "@pulumi/pulumi";
+import * as dynamic from "@pulumi/pulumi/dynamic";
+
+export class Provider implements dynamic.ResourceProvider {
```

Comment:
Curious did you need a custom provider here? No existing resource would do a trick I guess? TO fault-inject?